internal/file/spec.go

```diff
@@ -0,0 +1,29 @@
+package file
+
+const (
+	SpecTypeGPGKey FileSpecType = "gpgkey"
+)
+
+var fileSpecTypes = map[FileSpecType]struct{}{
+	SpecTypeGPGKey: {},
+}
+
+func (t FileSpecType) IsValid() bool {
+	_, ok := fileSpecTypes[t]
+	return ok
+}
+
+type FileSpecType string
+
+type InputSpec struct {
+	URL      string
+	Filename string
+	Type     FileSpecType
+}
+
+type Spec struct {
+	URL      string
+	Filename string
+	Checksum string
+	Type     FileSpecType
+}
```

Comment:
Can we get away with having just one spec?  Though I wonder if we even need a spec at all?  Looking at the resolve function, it might be simpler to just have the URL as an argument and the checksum as output.  The caller can group the checksum with the filename and URL after the fact (from outside) if necessary.  Though maybe it's convenient to have these all bundled in the job.

Reply:
Yeah absolutely. I updated it to a single spec... I misread the second part of your comment. We could also group the URL and checksum on the caller's side - we'd still need to add that to some kind of struct and then pass that down the chain to the manifest call though, right?

Reply:
I think that getting the content may be useful and we can always calculate checksum once we have it. returning just the checksum seems a bit of a oversimplification to me, but I would be OK with it if everyone finds it as a better approach. +1 to the rest...

Reply:
Ah yeah good point, we may still need the contents of the file. I will update that and we can calculate the checksum at a later stage, like you say

Reply:
Oh yeah brilliant - I hadn't thought of that, but it would make it easier to import the gpgkey in general. I'm going to push the changes now, just fixing up some of the tests.

Reply:
I've left the contents in `[]byte`, not sure if we want to return a string or keep the contents in bytes. Let me know

Reply:
`[]byte` is IMO preferred, because it is the most generic representation. We may end up fetching also binary files (not that I know about any requirement). It is simple to convert `[]byte` to `string`...